internal/services/cognitive/cognitive_account_ai_services_resource.go

```diff
@@ -0,0 +1,704 @@
+
+		"sku_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"F0", "F1", "S0", "S", "S1", "S2", "S3", "S4", "S5", "S6", "P0", "P1", "P2", "E0", "DC0",
+			}, false),
+		},
+
+		"custom_subdomain_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"customer_managed_key": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"key_vault_key_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: keyVaultValidate.NestedItemIdWithOptionalVersion,
+					},
+
+					"identity_client_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsUUID,
+					},
+				},
+			},
+		},
+
+		"fqdns": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
+
+		"local_auth_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"network_acls": {
+			Type:         pluginsdk.TypeList,
+			Optional:     true,
+			MaxItems:     1,
+			RequiredWith: []string{"custom_subdomain_name"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"default_action": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(cognitiveservicesaccounts.NetworkRuleActionAllow),
+							string(cognitiveservicesaccounts.NetworkRuleActionDeny),
+						}, false),
+					},
+					"ip_rules": {
+						Type:     pluginsdk.TypeSet,
+						Optional: true,
+						Set:      set.HashIPv4AddressOrCIDR,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+							ValidateFunc: validation.Any(
+								commonValidate.IPv4Address,
+								commonValidate.CIDR,
+							),
+						},
+					},
+
+					"virtual_network_rules": {
+						Type:       pluginsdk.TypeSet,
+						Optional:   true,
+						ConfigMode: pluginsdk.SchemaConfigModeAuto,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"subnet_id": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+								},
+
+								"ignore_missing_vnet_service_endpoint": {
+									Type:     pluginsdk.TypeBool,
+									Optional: true,
+									Default:  false,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+
+		"outbound_network_access_restricted": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"public_network_access_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
```

Comment:
rename to `public_network_access` ? Sorry, I didn't get the point.

Reply:
We're moving away from using `public_network_access_enabled`, since the Azure API is gradually supporting three values for this field (`Disabled`, `Enabled` and `SecuredByPerimeter`) - as such new resources should expose this as a string field (we'll be making this a `commonschema` field in https://github.com/hashicorp/go-azure-helpers/pull/238) - and existing resources will be updated in time.

As such, can we make this a string field, `public_network_access`, with the constant values being the possible values - rather than a boolean?

Reply:
`public_network_access_enabled` has been renamed to `public_network_access`